packages/app-store/shimmervideo/lib/VideoApiAdapter.ts

```diff
@@ -0,0 +1,195 @@
+import { z } from "zod";
+
+import { handleErrorsJson } from "@calcom/lib/errors";
+import type { GetRecordingsResponseSchema, GetAccessLinkResponseSchema } from "@calcom/prisma/zod-utils";
+import { getRecordingsResponseSchema, getAccessLinkResponseSchema } from "@calcom/prisma/zod-utils";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+import type { PartialReference } from "@calcom/types/EventManager";
+import type { VideoApiAdapter, VideoCallData } from "@calcom/types/VideoApiAdapter";
+
+import { getShimmerAppKeys } from "./getShimmerAppKeys";
+
+/** Shimmer Video app type in the config.json
+ *  changed to 'shimmer_video' to support video conferencing
+ */
+
+/** @link https://docs.daily.co/reference/rest-api/rooms/create-room */
+const dailyReturnTypeSchema = z.object({
+  /** Long UID string ie: 987b5eb5-d116-4a4e-8e2c-14fcb5710966 */
+  id: z.string(),
+  /** Not a real name, just a random generated string ie: "ePR84NQ1bPigp79dDezz" */
+  name: z.string(),
+  api_created: z.boolean(),
+  privacy: z.union([z.literal("private"), z.literal("public")]),
+  /** https://api-demo.daily.co/ePR84NQ1bPigp79dDezz */
+  url: z.string(),
+  created_at: z.string(),
+  config: z.object({
+    enable_prejoin_ui: z.boolean(),
+    enable_people_ui: z.boolean(),
+    enable_emoji_reactions: z.boolean(),
+    enable_pip_ui: z.boolean(),
+    enable_hand_raising: z.boolean(),
+    enable_network_ui: z.boolean(),
+    enable_video_processing_ui: z.boolean(),
+    enable_noise_cancellation_ui: z.boolean(),
+    enable_advanced_chat: z.boolean(),
+    //above flags are for prebuilt daily
+    enable_chat: z.boolean(),
+    enable_knocking: z.boolean(),
+  }),
+});
+
+export interface DailyEventResult {
+  id: string;
+  name: string;
+  api_created: boolean;
+  privacy: string;
+  url: string;
+  created_at: string;
+  config: Record<string, unknown>;
+}
+
+export interface DailyVideoCallData {
+  type: string;
+  id: string;
+  password: string;
+  url: string;
+}
+
+export const fetcher = async (endpoint: string, init?: RequestInit | undefined) => {
+  const { api_key } = await getShimmerAppKeys();
+  const response = await fetch(`https://api.daily.co/v1${endpoint}`, {
```

Comment:
Why are we hitting daily API from shimmer video app. If shimmer video app uses daily itself, it shouldn't depend on we hitting daily API. We should contact shimmer app only for that. Shimmer app can do that same at it's backend